### PR TITLE
#378 SCH服务中心开始watch模式后，报错空指针异常问题

### PR DESCRIPTION
**Pull request by develpoerX**

### CODE REVIEW

File: spring-cloud-huawei-common/src/main/java/com/huaweicloud/common/transport/ServiceCombSSLProperties.java [hunk 1]
```diff
@@ -83,9 +107,106 @@ public void setKeyStoreValue(String keyStoreValue) {
...
+  }
+
+  public void setCrl(String crl) {
+    this.crl = crl;
+  }
+
   public boolean isEmpty() {
     return StringUtils.isEmpty(trustStoreValue) || StringUtils.isEmpty(keyStoreValue) || StringUtils
-        .isEmpty(trustStore) || StringUtils.isEmpty(keyStore);
+        .isEmpty(trustStore) || StringUtils.isEmpty(keyStore) || StringUtils.isEmpty(sslCustomClass)
```
Comment (liubao68) [inline spring-cloud-huawei-common/src/main/java/com/huaweicloud/common/transport/ServiceCombSSLProperties.java:208]: do not modify isEmpty method. This used by old logic (config center client), which is not refactor yet.

File: spring-cloud-huawei-servicecomb-discovery/src/main/java/com/huaweicloud/servicecomb/discovery/DiscoveryAutoConfiguration.java [hunk 2]
```diff
@@ -79,17 +79,40 @@ private SSLProperties createSSLProperties(AddressManager addressManager,
...
-    sslOption.setTrustStoreValue(serviceCombSSLProperties.getTrustStoreValue());
+    sslOption.setKeyStoreType(serviceCombSSLProperties.getKeyStoreType() == null ?
+        SSLOption.DEFAULT_OPTION.getKeyStoreType() : serviceCombSSLProperties.getKeyStoreType().name());
+    sslOption.setKeyStore(serviceCombSSLProperties.getKeyStore() == null?
+        SSLOption.DEFAULT_OPTION.getKeyStore() : serviceCombSSLProperties.getKeyStore());
+    sslOption.setKeyStoreValue(serviceCombSSLProperties.getKeyStoreValue() == null?
+        SSLOption.DEFAULT_OPTION.getKeyStoreValue() : serviceCombSSLProperties.getKeyStoreValue());
+    sslOption.setTrustStoreType(ServiceCombSSLProperties.KeyStoreInstanceType.JKS.name() == null?
+        SSLOption.DEFAULT_OPTION.getTrustStoreType() : ServiceCombSSLProperties.KeyStoreInstanceType.JKS.name());
+    sslOption.setTrustStore(serviceCombSSLProperties.getTrustStore() == null?
```
Comment (liubao68) [inline spring-cloud-huawei-servicecomb-discovery/src/main/java/com/huaweicloud/servicecomb/discovery/DiscoveryAutoConfiguration.java:90]: JKS is a constant, can not be null

File: spring-cloud-huawei-servicecomb-discovery/src/main/java/com/huaweicloud/servicecomb/discovery/DiscoveryAutoConfiguration.java [hunk 3]
```diff
@@ -79,17 +79,40 @@ private SSLProperties createSSLProperties(AddressManager addressManager,
...
+        SSLOption.DEFAULT_OPTION.getSslCustomClass() : serviceCombSSLProperties.getSslCustomClass());
+    sslOption.setAuthPeer(serviceCombSSLProperties.isAuthPeer());
+    sslOption.setCheckCNHost(serviceCombSSLProperties.isCheckCNHost());
+    sslOption.setAllowRenegociate(serviceCombSSLProperties.isAllowRenegociate());
+    sslOption.setCheckCNWhite(serviceCombSSLProperties.isCheckCNWhite());
 
     sslProperties.setSslOption(sslOption);
     // TODO: support ssl password encryption
-    sslProperties.setSslCustom(SSLCustom.defaultSSLCustom());
+    sslProperties.setSslCustom(SSLCustom.createSSLCustom(serviceCombSSLProperties.getSslCustomClass()));
```
Comment (liubao68) [inline spring-cloud-huawei-servicecomb-discovery/src/main/java/com/huaweicloud/servicecomb/discovery/DiscoveryAutoConfiguration.java:115]: delete TODO comment

File: spring-cloud-huawei-common/src/main/java/com/huaweicloud/common/transport/ServiceCombSSLProperties.java (hunk 1 above)
Comment (develpoerX) [reply spring-cloud-huawei-common/src/main/java/com/huaweicloud/common/transport/ServiceCombSSLProperties.java]: done

File: spring-cloud-huawei-servicecomb-discovery/src/main/java/com/huaweicloud/servicecomb/discovery/DiscoveryAutoConfiguration.java (hunk 2 above)
Comment (develpoerX) [reply spring-cloud-huawei-servicecomb-discovery/src/main/java/com/huaweicloud/servicecomb/discovery/DiscoveryAutoConfiguration.java:90]: done.I have modify this to
```
SSLOption.DEFAULT_OPTION.getTrustStoreType() == null?
```

File: spring-cloud-huawei-servicecomb-discovery/src/main/java/com/huaweicloud/servicecomb/discovery/DiscoveryAutoConfiguration.java (hunk 3 above)
Comment (develpoerX) [reply spring-cloud-huawei-servicecomb-discovery/src/main/java/com/huaweicloud/servicecomb/discovery/DiscoveryAutoConfiguration.java:115]: done

**Review by liubao68 — APPROVED** [review]

**Pull request merged**